Apps/GatewayApi/src/Services/CommentService.cs

```diff
@@ -101,34 +98,21 @@ public RequestResult<IEnumerable<UserComment>> GetEntryComments(string hdId, str
             if (key == null)
             {
                 this.logger.LogError("User does not have a key: {HdId}", hdId);
-                return new RequestResult<IEnumerable<UserComment>>
-                {
-                    ResultStatus = ResultType.Error,
-                    ResultError = new RequestResultError
-                    {
-                        ResultMessage = "Profile Key not set",
-                        ErrorCode = ErrorTranslator.InternalError(ErrorType.InvalidState),
-                    },
-                };
+                return RequestResultFactory.ServiceError<IEnumerable<UserComment>>(ErrorType.InvalidState, ServiceType.Database, "Profile Key not set");
             }
 
             DbResult<IEnumerable<Comment>> dbComments = this.commentDelegate.GetByParentEntry(hdId, parentEntryId);
-            RequestResult<IEnumerable<UserComment>> result = new()
+
+            if (dbComments.Status != DbStatusCode.Read)
             {
-                ResourcePayload = dbComments.Payload.Select(c => CommentMapUtils.CreateFromDbModel(c, this.cryptoDelegate, key, this.autoMapper)),
-                TotalResultCount = dbComments.Payload.Count(),
-                PageIndex = 0,
-                PageSize = dbComments.Payload.Count(),
-                ResultStatus = dbComments.Status == DbStatusCode.Read ? ResultType.Success : ResultType.Error,
-                ResultError = dbComments.Status != DbStatusCode.Read
-                    ? new RequestResultError
-                    {
-                        ResultMessage = dbComments.Message,
-                        ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationInternal, ServiceType.Database),
-                    }
-                    : null,
-            };
-            return result;
+                return RequestResultFactory.ServiceError<IEnumerable<UserComment>>(ErrorType.CommunicationInternal, ServiceType.Database, dbComments.Message);
+            }
+
+            return RequestResultFactory.Success(
+                dbComments.Payload.Select(c => CommentMapUtils.CreateFromDbModel(c, this.cryptoDelegate, key, this.autoMapper)),
+                dbComments.Payload.Count(),
+                0,
+                dbComments.Payload.Count());
```

Comment:
I suggest updating the return value of GetByParentEntry in ICommentDelegate/DbCommentDelegate to `DbResult<IList<Comment>>` to avoid the appearance of and potential warnings about possible multiple enumerations.

Reply:
Changed